packages/docusaurus-theme-common/src/utils/navbarUtils.tsx

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {
+  type ReactNode,
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+} from 'react';
+import useWindowSize from '../hooks/useWindowSize';
+import {useHistoryPopHandler} from './historyUtils';
+import {NavbarSecondaryMenuProvider} from './navbarSecondaryMenuUtils';
+import {useActivePlugin} from '@docusaurus/plugin-content-docs/client';
+import {useThemeConfig} from './useThemeConfig';
+import {ReactContextError} from './reactUtils';
+
+const DefaultNavItemPosition = 'right';
+
+// If split links by left/right
+// if position is unspecified, fallback to right
+export function splitNavbarItems<
+  // TODO BAD, temporary type hack
+  T extends {position?: 'left' | 'right'},
```

Comment:
Why is this bad? It looks pretty sufficient to me

Reply:
If we didn't have the NavbarItem duplicate/incompatible types this wouldn't be necessary.

I'll keep the todo for now as IMHO we should only accept a NavbarItem type  (unless we have another reason to use structural typing to split other kinds of items? in which case this util should be made generic?)